File: surf-chat/src/main/java/dev/slne/surf/social/chat/provider/ChannelProvider.java
package dev.slne.surf.social.chat.provider;

import dev.slne.surf.social.chat.object.Channel;
import it.unimi.dsi.fastutil.objects.Object2ObjectMap;
import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
import java.util.UUID;
import lombok.Getter;

@Getter
public class ChannelProvider {
  @Getter
  private static final ChannelProvider instance = new ChannelProvider();
  private final Object2ObjectMap<UUID, Channel> channels = new Object2ObjectOpenHashMap<>();
}

File: surf-chat/src/main/java/dev/slne/surf/social/chat/provider/ChannelProvider.kt
package dev.slne.surf.social.chat.provider

import dev.slne.surf.social.chat.`object`.Channel
import dev.slne.surf.surfapi.core.api.util.mutableObject2ObjectMapOf
import org.bukkit.entity.Player
import java.util.*

object ChannelProvider {
    val channels = mutableObject2ObjectMapOf<UUID, Channel>()

    fun exists(name: String): Boolean = channels.values.any { it.name == name }

    fun handleQuit(player: Player) {
        val channel = Channel.getChannel(player) ?: return

        if (channel.owner == player) {
            channel.delete()
        }
    }
}
